man/man5/zfs-module-parameters.5

```diff
@@ -1752,14 +1769,97 @@ Default value: \fB2\fR.
 .ad
 .RS 12n
 Max concurrent I/Os per top-level vdev (mirrors or raidz arrays) allowed during
-scrub or resilver operations.
+legacy scrub or resilver operations.
 .sp
 Default value: \fB32\fR.
 .RE
 
 .sp
 .ne 2
 .na
+\fBzfs_scan_md_maxinflight\fR (int)
+.ad
+.RS 12n
+Max concurrent I/Os allowed during scrub or resilver metadata scanning phase.
+.sp
+Default value: \fB1024\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_leaf_maxinflight\fR (int)
+.ad
+.RS 12n
+Max concurrent I/Os per leaf vdev allowed during scrub or resilver data
+verification phase.
+.sp
+Default value: \fB50\fR.
```

Comment:
> Why should this be a different tunable that zfs_top_maxinflight?

So the idea here is that `zfs_top_maxinflight` would only be for legacy scrubs. The concept doesn't really make sense with the new scrubbing algorithm which does all data verification in parallel per top-level-vdev. Comparing a pool with 36 disks vs a pool with a single disk, I would want the one with 36 disks to take advantage of the full amount off extra bandwidth it has. That was the reason for turning the old global tunable into a per-leaf vdev tunable.

> And why should it be larger? I'd think that a much smaller value could be just as good, since the i/os you're issuing are already sorted (I think).

Again, I am very open to changing it. In this case, I don't think 50 outstanding IO's per disk is unreasonable, however.